### ISSUE TITLE
coq_makefile should error instead of trying to build plugins with the wrong OCaml

### ISSUE THREAD

**Issue by JasonGross** [coq/coq#12656]
I think `coq_makefile` should generate `Makefile`s that check the version of OCaml associated to `"$(OCAMLFIND)" opt` (or whatever) and if the version does not match the version used to compile Coq, rather than blindly calling `$(COQPP)` / `$(CAMLC)` / `$(CAMLOPTC)` / etc, it should instead spit out an error message along the lines of the one @andrew-appel suggests in my quote of him below.

-------------
Original context:

> I was just trying to compile metacoq, which is indeed a plugin.  Now that I understand better, I see in hindsight that much confusion (and time spent by several people) could have been avoided if only the following were true:
>
> coqpp should give an error message "Your ocaml configuration does not match the ocaml configuration under which coq was compiled.  Plug-ins can only be compiled using the same ocaml configuration as coq.  The only reliable way to achieve that is to compile coq yourself."
>
> This would have been much more explanatory.

_Originally posted by @andrew-appel in https://github.com/coq/coq/issues/10921#issuecomment-655116355_

**Comment by andrew-appel** [comment]
Well, there's more to it than that.  If coq_makefile is being run on a coq installation that was installed from precompiled binaries, then $(OCAMLFIND) itself is likely to be a nonexistent file-path...

**Comment by JasonGross** [comment]
Nah, I think that's fine.  If `$(OCAMLFIND)` is empty or non-existant, then using it to fetch the version number returns the empty string (possibly with an error message on stderr, which should be sent to /dev/null).  Errors in subshells aren't fatal for the parent shell, so we'll still get the error message, and something saying "Coq was compiled with OCaml version 'foo', your system has OCaml version ''" seems informative enough.  (Of course, we can choose to give a different error message when the version is empty.)
